geyser-plugin-manager/src/geyser_plugin_service.rs

```diff
@@ -122,70 +128,77 @@ impl GeyserPluginService {
             (None, None)
         };
 
+        let rpc_handler_thread = rpc_to_plugin_manager_receiver.map(|request_receiver| {
+            let rpc_plugin_manager = plugin_manager.clone();
+            Arc::new(std::thread::spawn(move || loop {
+                std::thread::park();
+                info!("we've been awakened");
+                // Wait 3 seconds to receive message after being awoken
+                if let Ok(request) = request_receiver.recv_timeout(Duration::from_secs(3)) {
+                    match request {
+                        PluginManagerRequest::ListPlugins { response_sender } => {
+                            let plugin_manager = rpc_plugin_manager.read().unwrap();
+                            response_sender
+                                .send(Self::list_plugins_rpc(&plugin_manager))
+                                .expect("admin rpc service will be waiting for response");
```

Comment:
Idiomatically, an `expect` message is supposed to contain a message explaining why it should never fail, not that it failed. See [this](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect)

In this case, the send will succeed because rpc service will not drop the receiver and will be waiting for the response.